[Entitlements] Small docs fixes (#127323)
elastic/elasticsearch: docs/extend/creating-classic-plugins.md

#### `manage_threads`

Allows code to call methods that create or modify properties on Java Threads, for example `Thread#start` or `ThreadGroup#setMaxPriority`.

> NOTE: in general this entitlement should *not* be needed! Your plugin should use {{es}} thread pools and executors (see `Plugin#getExecutorBuilders`) instead of creating and managing its own threads. Setting the name, priority, daemon state and context class loader are things no plugins should do when executing on
> {{es}} threadpools.
> 
> However, many 3rd party libraries that support async operations (e.g. Apache HTTP client) need to create and manage their own threads. In this case it is justifiable to request this entitlement.

Reviewer: ```suggestion
:::{note}
This entitlement is rarely necessary. Your plugin should use {{es}} thread pools and executors (see `Plugin#getExecutorBuilders`) instead of creating and managing its own threads. Setting the name, priority, daemon state and context class loader are things no plugins should do when executing on {{es}} threadpools.

However, many 3rd party libraries that support async operations (e.g. Apache HTTP client) need to create and manage their own threads. In this case it is justifiable to request this entitlement.
:::
```

Reviewer: If you want a true [admonition block](https://elastic.github.io/docs-builder/syntax/admonitions/#note) :)

Author: TIL
That's great @leemthompo, thank you very much!
